### PR TITLE
869: change order of upcoming tab milestones list display

### LINKED ISSUE #869: Upcoming milestone timeline display issues

### PR DESCRIPTION
**Pull request by trbmcginnis**
### Changes
Update order of milestones on the upcoming tab to match what's outlined in this issue: #869 

(1) first milestone in the project's list of milestones
(2) the "view full timeline" link
(3) the four milestones associated with reviews

### Tests
- added one new acceptance test for checking order of timeline on upcoming tab
- TODO: add two more tests for checking order of milestones on reviewed and archive tabs AND write new tests to account for more edge cases

Closes #869

### CODE REVIEW

File: tests/acceptance/milestone-timeline-on-tabs-displays-correctly-test.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    });
+  });
+
+  hooks.afterEach(async function() {
+    window.location.hash = '';
+
+    await invalidateSession();
+  });
+
+  test('milestones list on UPCOMING tab shows up in correct order', async function(assert) {
```
Comment (allthesignals) [inline tests/acceptance/milestone-timeline-on-tabs-displays-correctly-test.js]: YES thank you

File: app/models/assignment.js [hunk 2]
```diff
@@ -89,37 +89,25 @@ export default class AssignmentModel extends Model {
...
+    // first milestone in the list
     const [firstMilestone] = milestones;
 
-    // all milestones after last completed and up to and including the LUP's relevant milestone
-    const lastCompletedPosition = milestones.findIndex(milestone => milestone.id === lastCompletedMilestone.id);
-    const assigneeRelevantMilestonePosition = milestones.findIndex(milestone => milestone.dcpMilestone === assigneeMilestoneIdentifier);
-    const remainingMilestones = milestones.slice(lastCompletedPosition + 1, assigneeRelevantMilestonePosition + 1);
+    // review milestones refer to four milestones: App Reviewed at CPC Review Session, CB Review, BP Review, and BB Review
+    const { reviewMilestoneIds } = this.milestoneConstants;
+    const reviewMilestones = milestones.filter(milestone => reviewMilestoneIds.includes(milestone.dcpMilestone));
```
Comment (allthesignals) [inline app/models/assignment.js]: Great
